crates/swc_cli/src/commands/plugin.rs

```diff
@@ -0,0 +1,234 @@
+                }
+                out.push_str(item);
+                out.push('\n');
+            }
+
+            out
+        }
+    };
+
+    paths::append(&ignore_file_path, ignore.as_bytes())?;
+    Ok(())
+}
+
+impl super::CommandRunner for PluginScaffoldOptions {
+    /// Create a rust project for the plugin from template.
+    /// This largely mimic https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_new.rs,
+    /// but also thinner implementation based on some assumptions like skipping
+    /// to support non-git based vcs.
+    fn execute(&self) -> Result<()> {
+        let path = &self.path;
+        if path.exists() {
+            anyhow::bail!("destination `{}` already exists", path.display())
+        }
+
+        let name = get_name(self)?;
+
+        // Choose to rely on system's git binary instead of depends on git lib for now
+        // to avoid bring in large / heavy dependencies into cli binaries.
+        // Depends on our usecase grows, we can revisit this.
+        let mut base_git_cmd = if cfg!(target_os = "windows") {
+            let mut c = std::process::Command::new("cmd");
+            c.arg("/C").arg("git");
+            c
+        } else {
+            std::process::Command::new("git")
+        };
+
+        // init git repo
+        base_git_cmd
+            .args(["init", name])
+            .output()
+            .context("failed to create dir for the plugin")?;
+
+        // generate .gitignore
+        write_ignore_file(path)?;
+
+        // Create `Cargo.toml` file with necessary sections
+        paths::write(
+            &path.join("Cargo.toml"),
+            format!(
+                r#"[package]
+name = "{}"
+version = "0.1.0"
+edition = "2018"
+
+[lib]
+crate-type = ["cdylib"]
+
+# Workaround for https://github.com/Amanieu/parking_lot/issues/269
+# (swc_atoms -> string_cache -> parking_lot)
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+parking_lot_core = "=0.8.0"
+
+[dependencies]
+serde = "1"
+swc_plugin = "*""#,
+                name
+            )
+            .as_bytes(),
+        )?;
+
+        let build_target = match self.target_type {
+            PluginTargetType::Wasm32UnknownUnknown => "wasm32-unknown-unknown",
+            PluginTargetType::Wasm32Wasi => "wasm32-wasi",
+        };
+
+        // Create cargo config for build target
+        let cargo_config_path = path.join(".cargo");
+        create_dir_all(&cargo_config_path)?;
+        paths::write(
+            &cargo_config_path.join("config"),
+            format!(
+                r#"[build]
+target = "{}""#,
+                build_target
+            )
+            .as_bytes(),
+        )?;
+
+        // Create entrypoint src file
+        let src_path = path.join("src");
+        create_dir_all(&src_path)?;
+        paths::write(
+            &src_path.join("lib.rs"),
+            r#"use swc_plugin::{ast::*, plugin_transform};
+
+pub struct TransformVisitor;
+
+impl VisitMut for TransformVisitor {
```

Comment:
We may reexport `swc_common::util::Take` as VisitMut requires it.

Reply:
yes, added. Exposing / adding new one should be easy that we can do over iterations.